climate.py

```diff
@@ -321,8 +355,75 @@ def get_scouting_observation(token, api_key, scouting_observation_id):
 
     res = requests.get(uri, headers=headers)
     Logger().info(to_curl(res.request))
-    
+
     if res.status_code == 200:
         return res.json()
-    else:
-        return None
+
+    return None
+
+
+def get_scouting_observation_attachments(token,
+                                         api_key,
+                                         scouting_observation_id):
+    """
+    Retrieve attachments associated with a given scouting observation. Photos
+    added to scouting notes in the FieldView app are capped to 20MB, and we
+    won’t store photos larger than that in a scouting note.
+    https://dev.fieldview.com/technical-documentation/ for possible status
+    values and their meaning.
+    :param token: access_token
+    :param api_key: Provided by Climate
+    :param scouting_observation_id: Unique identifier of the
+        Scouting Observation.
+
+    """
+    uri = '{}/v4/layers/scoutingObservations/{}/attachments'.format(
+        api_uri, scouting_observation_id)
+    headers = {
+        'authorization': bearer_token(token),
+        'accept': json_content_type,
+        'x-api-key': api_key
+    }
+
+    res = requests.get(uri, headers=headers)
+    Logger().info(to_curl(res.request))
+
+    if res.status_code == 200:
+        return res.json()['results']
+
+    return []
```

Comment:
If the response code is 4XX or 5XX, does the python requests library raise an exception? If not, it looks like we'd be silently swallowing the error here (and probably in our other API calls too). It would be nice to log an error with the response code and body, so that it's clearer what happened.

Reply:
added code for logging errors on the server console.